main.c

```diff
@@ -316,6 +321,12 @@ STATIC bool run_code_py(safe_mode_t safe_mode) {
     }
     #endif
 
+    #ifdef CIRCUITPY_BOOT_COUNTER
```

Comment:
This isn't really a boot counter because you are incrementing every code.py run. How about putting it in `board_init()` instead? It's a small amount of code so I think it's ok to duplicate it in each board's `board_init`.

Reply:
@tannewt putting the counter in `board_init()` or `reset_board()` resulted in the counter being incremented by 2 sometimes 3 per reboot/restart/power cycle. I figured this was a result of various VM behavior?

Reply:
`board_init()` is called after fake sleep, as well as once in `main()`. `reset_board()` is called in `main()` and `cleanup_after_vm()`, so would be called twice (once for possible `boot.py` and once for `code.py`).

You could add your own routine just once to `main()` and conditionalize it with `CIRCUITPY_BOOT_COUNTER`, or something like that.

Reply:
Thanks @dhalbert! I'll move the `#ifdef CIRCUITPY_BOOT_COUNTER` to `main()`

Reply:
Yes, let's see if it works. BTW, use `#if` instead of `#ifdef`, and set it to `0` as the default. That is safer all around. We use that style for all `CIRCUITPY_*` flags.